File: JavaParser/If_Statement.java

public class If_Statement implements Statement
{
	private Boolean_Expression boolexpr1;
	
	private Compound cp1;
	
	private Compound cp2;
	
	public If_Statement()
	{	}
	
	public If_Statement(Boolean_Expression boolexpr1, Compound cp1, Compound cp2)
	{
		if (boolexpr1 == null)
			throw new IllegalArgumentException("boolexpr1 error /IfStatement");
		if (cp1 == null)
			throw new IllegalArgumentException("cp1 error /IfStatemetn");
		if (cp2 == null)
			throw new IllegalArgumentException("cp2 error /IfStatemetn");
		
		this.boolexpr1 = boolexpr1;
		this.cp1 = cp1;
		this.cp2 = cp2;
	}
	
	@Override
	public void execute()
	{
		// if boolexpression is true, then execute first code block else execute second code block
		if (boolexpr1.evaluate() == 1)
		{
			cp1.execute();
		}
		else if (boolexpr1.evaluate() == 0)
			
			/***********************
			 * just else
			 */
		{
			cp2.execute();
		}
	}
}

File: PythonParser/IfStatement.py
__author__ = 'ravipatel'

from Statement import Statement

class IfStatement(Statement):

    def __init__(self, boolexpr1, cp1, cp2):
        if (boolexpr1 is None):
            raise ValueError("boolexpr1 in ifstatement is wrong")
        self.boolexpr1 = boolexpr1
        if (cp1 is None):
            raise ValueError("cp1 in ifstatement is wrong")
        self.cp1 = cp1
        if (cp2 is None):
            raise ValueError("cp2 in ifstatement is wrong")
        self.cp2 = cp2

    def execute(self):
        if (self.boolexpr1.evaluate() is True):
            self.cp1.execute()
        elif (self.boolexpr1.evaluate() is False):
            self.cp2.execute()
            '''
            what if neither of the 2 (returns something besides a boolean) is true?
            '''